uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/toil-request-ctrl.spec.js

```diff
@@ -86,6 +87,7 @@
         spyOn(AbsenceTypeAPI, 'all').and.callThrough();
         spyOn(AbsenceTypeAPI, 'calculateToilExpiryDate').and.callThrough();
         spyOn(TOILRequestInstance, 'init').and.callThrough();
+        spyOn(AbsenceType, 'canExpire').and.returnValue($q.resolve(true));
```

Comment:
By default, the mock for `absenceTypeAPI.all` returns a list of records. The way `canExpire` works is to query the API using the absence type id and check if the expiration unit and duration are set to null. Since the mock returns a list of records canExpire will think that the query is true and the record never expires.

I tried to change the query so it returns the record only if the expiration unit and duration are not null, but it seems I can't make that kind of queries to the api. Or is there? I think it would make much more sense to change the query.